lib/resty/radixtree.lua

```diff
@@ -218,10 +219,12 @@ local function insert_route(self, opts)
     local data_idx = radix.radix_tree_find(self.tree, path, #path)
     if data_idx then
```

Comment:
We can check it with `data_idx ~= nil` instead. There is no need to check idx after this change.

Reply:
radix_tree_find never return nil, it only return 0 when not found or other exception occurrs

so check idx > 0 is enough in my consideration

Reply:
> radix_tree_find never return nil, it only return 0 when not found or other exception occurrs

No. You think it returns 0 because `tonumber(ffi_cast('intptr_t', data_idx))` is 0 when `data_idx` is nil.

Reply:
updated according to disscuss

Reply:
Actually we need to change `if data_idx then` to `if data_idx ~= nil then`. They are different things. You can read this for the details: https://segmentfault.com/a/1190000007937895

Reply:
good article, appreciated, ^_^